walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,134 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Iterator;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+
+    public List<TimeRange> findRange( ArrayList timeRangesArray, MeetingRequest request, Collection<Event> event){
+        int duration = (int)request.getDuration();
+        List<TimeRange> rangezz = new ArrayList<TimeRange>();
+        int timeRange = 0;
+        int timePlus = 0;
+        for(int i = 0; i < timeRangesArray.size(); i++) {
+            timeRange = (int)timeRangesArray.get(i);
+            // if there are duplicates remove those instances 
+            if (i == 0) {  
+                // skip if curr time = Start of Day (0)
+                if (timeRange == TimeRange.START_OF_DAY) {
+                    continue;
+                }
+                rangezz.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, timeRange, false));
+            }
+            else if (i == (timeRangesArray.size() - 1)) {
+                // skip if curr time = End of Day (1440)
+                if (timeRange == (TimeRange.END_OF_DAY + 1)) {
+                    break;
+                }
+                rangezz.add(TimeRange.fromStartEnd(timeRange, TimeRange.END_OF_DAY, true));
+            }
+            else {
+                timePlus = (int)timeRangesArray.get(++i);
+                // checks for duplicate times if we have duplicates just skip
+                if (timeRange == timePlus) {
+                    continue;
+                }
+                rangezz.add(TimeRange.fromStartEnd(timeRange, timePlus, false));                
+            }
+        }
+        for(int i = 0; i < rangezz.size(); i++) {
+            // checks to see if there if enough room in the day for each meeting if not remove it
+            if (rangezz.get(i).duration() < request.getDuration()) {
+                rangezz.remove(i);
+            }
+        }
+        return rangezz;
+    }
+
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+                return Arrays.asList();
+            }
+        if (events.isEmpty()) {
+            return Arrays.asList(TimeRange.WHOLE_DAY);
+        }
+        int eventCount = 0;
+        List<TimeRange> avaiablity = new ArrayList<TimeRange>();
+        ArrayList<Integer>  startEndTimes = new ArrayList<Integer>();
+        int start = 0;
+        int end = 0;
+        int startTimePrev = 0;
+        int endTimePrev = 0;
+        int index = 0;
+        for (Event e : events) {            
```

Comment:
Single letter variables are somewhat controversial, particularly when they're used in a fairly large block like this. Consider naming this "event" instead of just "e".